nuxeo-opencmis-impl/src/main/java/org/nuxeo/ecm/core/opencmis/impl/server/NuxeoPropertyData.java

```diff
@@ -346,6 +355,73 @@ public static Blob getPersistentBlob(ContentStream contentStream, String filenam
         return Blobs.createBlob(file, contentStream.getMimeType(), null, filename);
     }
 
+    public static void validateBlobDigest(Blob blob, HttpServletRequest request) {
+        if (blob == null) {
+            return;
+        }
+        String digestAlgorithm = blob.getDigestAlgorithm();
+        if (digestAlgorithm == null) {
+            return;
+        }
+        String reqDigest = NuxeoPropertyData.extractDigestFromRequestHeaders(request, digestAlgorithm);
+        if (reqDigest == null) {
+            return;
+        }
+        String blobDigest = blob.getDigest();
+        if (!blobDigest.equals(reqDigest)) {
+            throw new CmisInvalidArgumentException(String.format(
+                    "Content Stream Hex-encoded Digest: '%s' must equal Request Header Hex-encoded Digest: '%s'",
+                    blobDigest, reqDigest));
+        }
+    }
+
+    protected static String extractDigestFromRequestHeaders(HttpServletRequest request, String digestAlgorithm) {
+        if (request == null) {
+            return null;
+        }
+        if (request instanceof HttpServletRequestWrapper) {
+            request = (HttpServletRequest) ((HttpServletRequestWrapper) request).getRequest();
+        }
+        Enumeration<String> digests = request.getHeaders("digest");
+        if (digests == null) {
+            return null;
+        }
+        String digestAlgorithmUC = digestAlgorithm.toUpperCase();
+        Iterator it = Iterators.forEnumeration(digests);
+        while (it.hasNext()) {
+            String digest = (String) it.next();
+            int equals = digest.indexOf('=');
+            if (equals < 0) {
+                continue;
+            }
+            String reqDigestAlgorithmUC = digest.substring(0, equals).toUpperCase();
+            switch (reqDigestAlgorithmUC) {
+            case AbstractBinaryManager.MD5_DIGEST:
+            case AbstractBinaryManager.SHA1_DIGEST:
+            case AbstractBinaryManager.SHA256_DIGEST:
+                break;
+            default:
+                continue;
+            }
+            digest = digest.substring(equals + 1);
+            digest = transcodeBase64ToHex(digest);
+            return digest;
+        }
+        return null;
+    }
+
+    public static String transcodeBase64ToHex(String base64String){
+        byte[] bytes = Base64.decodeBase64(base64String);
+        String hexString = Hex.encodeHexString(bytes);
+        return hexString;
+    }
+
+    public static String transcodeHexToBase64(String hexString) throws DecoderException {
```

Comment:
As mentioned before, deal with `DecoderException` inside the method, don't expose it to callers.